cpp/splinepy/py/py_spline.hpp

```diff
@@ -373,6 +373,36 @@ class PySpline {
     return cmr;
   }
 
+  /// @brief Evaluate Splines at boundary face centers
+  /// @return numpy array with results
+  py::array_t<double> EvaluateBoundaryCenters() const {
+    // prepare output
+    const int n_faces = 2 * para_dim_;
+    py::array_t<double> queries(n_faces * para_dim_);
+    double* queries_ptr = static_cast<double*>(queries.request().ptr);
+    py::array_t<double> face_centers(n_faces * dim_);
+    double* face_centers_ptr = static_cast<double*>(face_centers.request().ptr);
```

Comment:
This works, but you can also take a bit lightweight approach with `std::vector<double> queries(...);` + `queries.data()`